### PR TITLE
PERF-2632: Add performance test for $fill in time series collections

### PR DESCRIPTION
**Pull request by madelinezec**

### CODE REVIEW

**Comment by madelinezec** [comment]
There is a failure in the auto-generated evergreen build, but it looks to be related to this bug https://jira.mongodb.org/browse/TIG-3236 so I believe we can ignore

**Comment by apodelko** [comment]
Did you run a patch with this workload? If yes, please share a link. I believe that it is important to make sure that it works properly in sys-perf. Also it would be good to run it several times to make sure that we don't have high variability.

**Comment by apodelko** [comment]
Also, to make sure, you want to run it on Standalone and 3-Node ReplSet for master only?

**Comment by madelinezec** [comment]
> Also, to make sure, you want to run it on Standalone and 3-Node ReplSet for master only?

I think we could definitely use product performance team's expertise re which build variants we should run this on for tests. It is my understanding that Standalone, 1-node replset, and 3-node replset are the most basic, so that's why I chose them. We definitely only want to run this for master.

File: src/workloads/execution/FillTimeseriesCollection.yml [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      OperationCommand:
+        aggregate: *coll
+        pipeline: [{
+          $fill: {
+            partitionBy: {part1: "$part1", part2: "$part2"},
+            sortBy: {timestamp: 1},
+            output: {
+              numeric: {
+                method: "linear"},
+              double: {
```
Comment (TedTuckman) [inline src/workloads/execution/FillTimeseriesCollection.yml:217]: Should one of these be "fill"?

**Review by TedTuckman — APPROVED** [review]
Generally happy with this, one comment

File: src/workloads/execution/FillTimeseriesCollection.yml [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+                method: "linear"},
```
[the remark this answers is not in the document]
Comment (madelinezec) [reply src/workloads/execution/FillTimeseriesCollection.yml:218]: Oh that's an interesting idea. I'll add it.

**Comment by apodelko** [comment]
> Standalone, 1-node replset, and 3-node replset 

Makes sense for me. Then you may add '- single-replica'. It would be good to run all of them as a patch to see what kind of results we are getting.

**Review by apodelko — APPROVED** [review]
LGTM

FYI only: Here is an answer for a strange thing I noticed - there’s an open bug where the timeseries collection inserts and queries don’t show in ftdc (see https://jira.mongodb.org/browse/SERVER-60105)

**Comment by apodelko** [comment]
And I assume that you run the patch before adding single-replica. And you selected '3-Node ReplicaSet (SSL)' - which is different from 'replica'. I believe it is 'replica-ssl' - so we don't see that genny workload there either. I guess unless you have specific reasons for SSL, we may stick to the original selection.

**Comment by madelinezec** [comment]
> And I assume that you run the patch before adding single-replica. And you selected '3-Node ReplicaSet (SSL)' - which is different from 'replica'. I believe it is 'replica-ssl' - so we don't see that genny workload there either. I guess unless you have specific reasons for SSL, we may stick to the original selection.

I ran the patch after adding single-replica

> And I assume that you run the patch before adding single-replica. And you selected '3-Node ReplicaSet (SSL)' - which is different from 'replica'. I believe it is 'replica-ssl' - so we don't see that genny workload there either. I guess unless you have specific reasons for SSL, we may stick to the original selection.

I have some more patches currently running after adding `- single-replica`, I'll share those with you as well when they finish. This won't actually get merged before mid next week.

**Pull request merged**